src/fides/api/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -573,3 +575,47 @@ def delete_dataset(
         "Deleting dataset '{}' for connection '{}'", fides_key, connection_config.key
     )
     dataset_config.delete(db)
+
+
+@router.get(
+    f"/filter{DATASETS}",
```

Comment:
I needed to create a new route for this because the current `/dataset` route is using the ctl autogenerated CRUD routes that aren't easy to change. This is a temporary route until those CRUD routes are refactored away.

I used  `filter` as a prefix because `/datasets/<fides_key>` is already a route so I can't do something like `/datasets/filter` because it matches that route